3-UsingAPIs/Week2/assignment/ex2-pokemonApp/index.js

```diff
@@ -22,18 +23,88 @@ Use async/await and try/catch to handle promises.
 Try and avoid using global variables. As much as possible, try and use function 
 parameters and return values to pass data back and forth.
 ------------------------------------------------------------------------------*/
-function fetchData(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+
+async function fetchData(url) {
+  try {
+    const response = await fetch(url);
+    if (!response.ok) {
+      throw new Error(`HTTP error! Status: ${response.status}`);
+    }
+    return await response.json();
+  } catch (error) {
+    console.error('Error fetching data:', error.message);
+  }
```

Comment:
Since you are catching the error here the try/catch blocks in the `fetchAndPopulatePokemons()` and `fetchImage()` functions will not see the error anymore: it is already "handled" here. If you want to console.log the error here and still allow for the calling functions to "see" the error too you can rethrow the error like this:

```suggestion
  } catch (error) {
    console.error('Error fetching data:', error.message);
    throw error; // <== this rethrows the error
  }
```